app/shared/actions/namebids.js

```diff
@@ -52,6 +51,62 @@ export function getBidForName(name) {
   };
 }
 
+export function getBidsForAccount(previous = false) {
+  return (dispatch: () => void, getState) => {
+    const { connection, settings } = getState();
+    let { recentBids } = settings;
+
+    const query = {
+      code: 'eosio',
+      json: true,
+      limit: 1000,
+      scope: 'eosio',
+      table: 'namebids'
+    };
+
+    if (previous) {
+      console.log({previous})
+      query.lower_bound = previous;
+    } else {
+      dispatch({
+        type: types.SYSTEM_NAMEBID_PENDING
+      });
+    }
+
+    eos(connection).getTableRows(query).then((results) => {
+      const { rows } = results;
+
+      rows.forEach((namebid) => {
+        const currentBids = get(recentBids, `${settings.chainId}.${settings.account}`) || [];
+        const bidIndex = findIndex(currentBids, { newname: namebid.newname });
+
+        if (bidIndex === -1 && namebid.high_bidder === settings.account) {
+          recentBids = set(recentBids, `${settings.chainId}.${settings.account}`, currentBids.concat({
+            newname: namebid.newname,
+            highestBid: `${(namebid.high_bid / 10000).toFixed(4)} EOS`
+          }));
+        }
+      });
+
+      dispatch(setSetting('recentBids', recentBids));
```

Comment:
Does this need to happen on every iteration of the loop, or could it just happen when `results.more` is false (line 97 below)?

It works, I just noticed it calling the `SET_SETTING` action repeatedly as it scanned for my bids :)

Reply:
Yeah I actually thought of that, but decided to go the current route since it will gradually add bids to the list as opposed to all at once at the end. It does end up calling `SET_SETTING` a lot though 😄

Reply:
Ah that makes sense, and I suppose it's not terrible since it sort of shows progress